algo.py

```diff
@@ -1,14 +1,15 @@
 from __future__ import division
-from sklearn.linear_model import LinearRegression, Ridge
-from sklearn.ensemble import RandomForestRegressor
-from sklearn.pipeline import Pipeline
+
 import re
-import pandas as pd
-from sklearn.feature_extraction.text import CountVectorizer, TfidfTransformer
+
+from pandas import DataFrame
```

Comment:
Are you using the from imports for external imports, and import MODULE_NAME for internal modules?  Is this a recommended style? (not questioning, just curious)